nmigen/build/run.py

```diff
@@ -162,3 +221,25 @@ def get(self, filename, mode="b"):
         super().get(filename, mode)
         with open(os.path.join(self.__root, filename), "r" + mode) as f:
             return f.read()
+
+
+class RemoteSshBuildProducts(BuildProducts):
+    def __init__(self, root, hostname, connect_args):
+        self.__root = root
+        self.__hostname = hostname
+        self.__connect_args = connect_args
+
+    def get(self, filename, mode="b"):
+        super().get(filename, mode)
+
+        from paramiko import SSHClient
+
+        with SSHClient() as client:
```

Comment:
Can you reuse the SSH client created earlier?

Reply:
Sure. But if I do that, do you object to a `run_script=False` input argument to `execute_remote_ssh`, so that a user is capable of skipping building remotely while _allowing_ to still program by grabbing remote build artifacts?

Reply:
Sure, that seems completely reasonable to have for parity with local builds.

Reply:
One caveat to reusing the SSH client is that I lose the advantage of a context manager to automatically close the client. Per http://docs.paramiko.org/en/stable/api/client.html#paramiko.client.SSHClient.close:

>Warning
Failure to [close] may, in some situations, cause your Python interpreter to hang at shutdown (often due to race conditions). It’s good practice to close your client objects anytime you’re done using them, instead of relying on garbage collection.

I need something that is guaranteed to call `close()`. AFAIU, adding ` __del__` to `RemoteSSHBuildProducts` won't help because it's not guaranteed `__del__` will run.

Reply:
Yes, that's true. I guess it's imperfect either way, so let's go with two separate client instances.